packages/@aws-cdk/aws-gamelift/lib/matchmaking-configuration.ts

```diff
@@ -0,0 +1,286 @@
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as sns from '@aws-cdk/aws-sns';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { IMatchmakingRuleSet } from '.';
+
+/**
+   * A set of custom properties for a game session, formatted as key-value pairs.
+   * These properties are passed to a game server process with a request to start a new game session.
+   *
+   * This parameter is not used for Standalone FlexMatch mode.
+   *
+   * @see https://docs.aws.amazon.com/gamelift/latest/developerguide/gamelift-sdk-server-api.html#gamelift-sdk-server-startsession
+   */
+export interface GameProperty {
+  /**
+     * The game property identifier.
+     */
+  readonly key: string;
+  /**
+     * The game property value.
+     */
+  readonly value: string;
+}
+
+/**
+ * Represents a Gamelift matchmaking configuration
+ */
+export interface IMatchmakingConfiguration extends cdk.IResource {
+  /**
+     * The name of the matchmaking configuration.
+     *
+     * @attribute
+     */
+  readonly matchmakingConfigurationName: string;
```

Comment:
I feel like there should be some sort of lib string construct that should satisfy this use case of fitting a regex pattern but I'm not sure it actually exists. Something like this but not just for ssm. https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ssm.StringParameter.html

I can't seem to find aything though in our general lib module. https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib-readme.html

I'll sync with some other engineers to check

Reply:
synced with romain. Yeah we don't have anything right now but he agrees there's probably value in us creating it